src/main/java/ToDoListFeature/ToDoList.java

```diff
@@ -33,83 +52,149 @@ private void unMarkTask(int index) {
         System.out.println("[mark X] Mark number X task as Done");
         System.out.println("[unmark X] Mark number X task as UnDone");
-        System.out.println("[quit] Return to main menu");
-        System.out.println("Any other words will be added as Todo tasks");
+        System.out.println("[todo XXX] Add a todo task");
+        System.out.println("[deadline XXX /by XXX] Add a deadline task");
+        System.out.println("[event XXX /from XXX /to XXX] Add a Event task");
         System.out.println("[help] Show commands");
+        System.out.println("[quit] Return to main menu");
         messageDivider();
         System.out.println("Please enter a command:");
     }
 
+    /**
+     * Print the message divider
+     */
     public static void messageDivider() {
         System.out.println("------------------------------------------");
     }
+    
+    /**
+     * Get the position of the first space in the input text
+     * @param inputText the input text
+     * @return the position of the first space in the input text
+     */
+    private int getNumberPosition(String inputText) {
+        return inputText.indexOf(" ");
+    }
+    
+    /**
+     * Process the keyword input and add the corresponding task to the list
+     * @param inputText the input text
+     * @return true if the input is a keyword, false otherwise
+     */
+    private boolean processKeywordInput(String inputText) {
+        int numberPosition = getNumberPosition(inputText);
+        if (matchesPattern(inputText, MARK_FEATURE_PATTERN)) {
+            markTask(Integer.parseInt(inputText.substring(numberPosition + SPACE_OFFSET)));
+            return true;
+        } else if (matchesPattern(inputText, UNMARK_FEATURE_PATTERN)) {
+            unMarkTask(Integer.parseInt(inputText.substring(numberPosition + SPACE_OFFSET)));
+            return true;
+        } else if (matchesPattern(inputText, TODOTASK_FEATURE_PATTERN)) {
+            addItem(new ToDoTask(inputText.substring(numberPosition + SPACE_OFFSET)));
+            return true;
+        } else if (matchesPattern(inputText, DEADLINESTASK_FEATURE_PATTERN)) {
+            int keywordPosition = inputText.indexOf("/by");
+            addItem(new DeadLinesTask(inputText.substring(numberPosition + SPACE_OFFSET, keywordPosition - 1),
+                    inputText.substring(keywordPosition + BY_OFFSET)));
+            return true;
+        } else if (matchesPattern(inputText, EVENTSTASK_PATTERN)) {
+            int keywordPosition1 = inputText.indexOf("/from"); // theck /from keyword position
+            int keywordPosition2 = inputText.indexOf("/to"); // check /to keyword position
+            addItem(new EventsTask(inputText.substring(numberPosition + SPACE_OFFSET, keywordPosition1 - 1),
+                    inputText.substring(keywordPosition1 + FROM_OFFSET, keywordPosition2 - 1),
+                    inputText.substring(keywordPosition2 + TO_OFFSET)));
```

Comment:
Perhaps can consider dealing with Strings using Matcher methods (since you are already using the regex package), which is more abstract than dealing them with String methods